Tests/ApolloCodegenTests/ApolloCodegenTests.swift

```diff
@@ -1912,8 +1914,172 @@ class ApolloCodegenTests: XCTestCase {
     ), rootURL: nil)
 
     // then
-    expect(try ApolloCodegen.validate(config: configContext))
+    expect(try ApolloCodegen.validate(config: configContext, compilationResult: .mock()))
       .to(throwError(ApolloCodegen.Error.inputSearchPathInvalid(path: "schema/*.")))
   }
 
+  // MARK: Validation Tests (conflicting schema name)
```

Comment:
@AnthonyMDev these tests do the job but they rely on knowing the implementation detail of how we name the generated files, i.e.: there is currently no test for mutations because we append "Mutation" or "Local CacheMutation" to the filename.

I wonder if these tests might be better suited to going in the file generators?

Reply:
I think this is appropriate here. Unit tests define an expected behavior. Because they are unit tests, _not_ integration tests, that sometimes relies on implementation details. As long a the production code doesn't rely on implementation details, I think this is where we verify that we aren't violating the behaviors as implementation details change in the future.

Just clarifying, there is also no validation for queries and subscriptions, just for schema types right? You specifically called out mutations which threw me off there.

Reply:
> Just clarifying, there is also no validation for queries and subscriptions, just for schema types right? You specifically called out mutations which threw me off there.

Schema types and fragments because we don't always append "Fragment" to the fragment filename. Operations should always have their _type_ appended to the filename.